bip-MarcoPon-BlockchainURI.mediawiki

```diff
@@ -0,0 +1,81 @@
+<pre>
+  BIP: bip-MarcoPon-BlockchainURI
+  Title: URI scheme for Blockchain references / exploration
+  Author: Marco Pontello <marcopon@gmail.com>
+  Status: Draft
+  Type: Standards Track
+  Created: 29 August 2015
+  Post-History: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2015-August/010712.html
+</pre>
+
+==Abstract==
+
+This BIP propose an URI scheme for looking up blocks, transactions, addresses on a Blockchain explorer, or in general to make proper Blockchain references
+
+==Motivation==
+
+The purpose of this URI scheme is to enable users to handle all the requests for details about blocks, transactions, etc. with their preferred tool (being that a web service or a local application).
+Currently a Bitcoin client usually points to an arbitrary blockchain explorer when the user looks for the details of a transaction, or let him choose from a limited list of possible alternatives.
+Other times resorting to cut&paste is needed.
+The same happens with posts and messages that reference some particular txs or blocks, if they provide links at all.
+
+==Specification==
+
+The URI follow this simple form:
+
+ <nowiki>blockchain:[//chain]</type/hash></nowiki>
+
+Where:
+
+;chain:
+: (optional) to uniquely point to a specific chain, the hash of the corresponding genesis block is used (leading zeros included). For forked chains, the hash of the relevant first block after fork is used. In principle some kind of alias/mnemonic could also be used, but that is out of the scope of this BIP, and maybe could be developed in another subsequent one. If omitted (which would be the usual case), Bitcoin's mainnet is assumed. As a reference, see for example this code fragment from Bitcoin chainparams.cpp source:
+  const std::map<std::string, uint256> CChainParams::supportedChains =
+      boost::assign::map_list_of
+      ( "main", uint256S("0x000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f"))
+      ( "test", uint256S("0x000000000933ea01ad0ee984209779baaec3ced90fa3f408719526f8d77f4943"))
+      ( "regtest", uint256S("0x0f9188f13cb7b2c71f2a335e3a4fc328bf5beb436012afca590b1a11466e2206"))
+      ;
+
+
+
+;type: 
+: '''tx''': for transactions.
+: '''block''': for blocks (supports both hash or height).
+: '''address''': for addresses.
+
+;hash: 
+: the relevant hash to refer to (leading zeros included), or block height.
+
+==Examples==
+
+A transaction on Bitcoin's mainnet:
+ blockchain:/tx/b462ae6eb8bdae2e060239a2a3ea5d9c3e0f9ef34d9717beb2dcf0ed42cee7da
+
+A block on Bitcoin's mainnet:
+ blockchain:/block/00000000000000000119af5bcae2926df54ae262e9071a94a99c913cc217cc72
+or
+ blockchain:/block/372338
+
+An address on Bitcoin's mainnet:
+ blockchain:/address/16EW6Rv9P9AxFDBrZV816dD4sj1EAYUX3f
```

Comment:
I used the items generally linked / used on block explorers, but I suppose it could be added. BTW, how about other items that may become relevant when segwit is implemented?